src/Compilers/Test/Utilities/CSharp/CSharpTestBase.cs

```diff
@@ -1250,12 +1250,19 @@ public static SyntaxTree ParseWithRoundTripCheck(string text, CSharpParseOptions
             return compilation;
         }
 
-        private static CSharpCompilationOptions CheckForTopLevelStatements(SyntaxTree[] syntaxTrees)
+        private static CSharpCompilationOptions CheckForTopLevelStatementsAndUnsafe(SyntaxTree[] syntaxTrees)
```

Comment:
At this point is it worth making the check a syntax walker that returns a flag enum? Yesterday it's tls, today it's tls and unsafe, tomorrow it's another new thing?

Reply:
Returning options directly is fine, what's the benefit of adding an extra layer of indirection?

Reply:
Avoid an extra syntax walk on nearly every syntax tree in nearly every compiler test, and avoiding allocations of new compilation options for unsafe tests.

Reply:
I see. I realized what you meant this afternoon. I was slow ;-) Sure!